lib/http/apps/dao.js

```diff
@@ -20,20 +24,50 @@ const app = express();
  * @apiUse InternalServerError
  */
 app.post(`/`, (req, res) => {
+  let spaceId = req.body.spaceId;
   let spaceName = req.body.spaceName;
   let descHash = req.body.descHash;
   let coinName = req.body.coinName;
   let coinSymbol = req.body.coinSymbol;
   let coinDecimals = req.body.coinDecimals;
 
   console.log(`New DAO: ${spaceName} ${descHash} ${coinName} ${coinSymbol} ${coinDecimals}  with root: ${res.locals.wallet.address}`);
-  res.locals.kit.newInstance({ spaceName, descHash, coinName, coinSymbol, coinDecimals, root: res.locals.wallet.address }, { gasLimit: 5000000 }).then((dao) => {
-    res.status(201).json(dao);
-  }).catch(e => {
-    console.log(e);
-    res.sendStatus(500);
+
+  res.locals.kit.newInstance({ spaceName, descHash, coinName, coinSymbol, coinDecimals, root: res.locals.wallet.address }, { gasLimit: 5000000 }).then(dao => {
+    saveDao(spaceId, JSON.stringify(dao));
```

Comment:
this will not work. now you seem to store the transaction response. 

this needs to be something like 

```js
newInstance({...}).then(transaction => {
  res.sendStatus(201);
  transaction.wait(2).then(result => {
    const txHash = result.transactionHash;
    const deployEvent = result.events.find(l => l.event === 'DeployInstance');
     //return apps proxy address
     const apps = result.events
        .filter(l => l.event === 'InstalledApp')
        .map(event => {
           return { id: event.args.appId, proxy: event.args.appProxy }
        })
      if (!deployEvent) { throw new Error('No DeployInstance event found') }
      saveDao({ txHash: txHash, daoAddress: deployEvent.args.dao, apps: apps});
  });
});
```

Reply:
yup my bad